models/status.go

```diff
@@ -145,9 +146,27 @@ func GetCommitStatuses(repo *Repository, sha string, page int) ([]*CommitStatus,
 // GetLatestCommitStatus returns all statuses with a unique context for a given commit.
 func GetLatestCommitStatus(repo *Repository, sha string, page int) ([]*CommitStatus, error) {
 	statuses := make([]*CommitStatus, 0, 10)
-	return statuses, x.Limit(10, page*10).
-		Where("repo_id = ?", repo.ID).And("sha = ?", sha).Select("*").
-		GroupBy("context").Desc("created_unix").Find(&statuses)
+	err := x.Limit(10, page*10).
+		Where("repo_id = ?", repo.ID).And("sha = ?", sha).
+		Select("context, max(created_unix) as created_unix").
+		GroupBy("context").OrderBy("max(created_unix) desc").Find(&statuses)
+	if err != nil {
+		return nil, err
+	}
+	fullStatuses := make([]*CommitStatus, 0, len(statuses))
+	for _, status := range statuses {
+		cs := make([]*CommitStatus, 0, 1)
+		err = x.Where("repo_id = ?", repo.ID).And("sha = ?", sha).
```

Comment:
Because there is no way to get newest row grouped by context in one select that would work across all supported databases